test/es-module/test-esm-loader-programmatically.mjs

```diff
@@ -182,26 +182,36 @@ describe('ESM: programmatically register loaders', { concurrency: true }, () =>
 
     const lines = stdout.split('\n');
 
+    // Resolve occurs twice because it is first used to resolve the `load` loader
+    // _AND THEN_ the `register` module.
     assert.match(lines[0], /resolve passthru/);
-    assert.match(lines[1], /load passthru/);
-    assert.match(lines[2], /Hello from dynamic import/);
+    assert.match(lines[1], /resolve passthru/);
+    assert.match(lines[2], /load passthru/);
+    assert.match(lines[3], /Hello from dynamic import/);
 
-    assert.strictEqual(lines[3], '');
+    assert.strictEqual(lines[4], '');
   });
 
-  it('does not work without dummy CLI loader', async () => {
+  it('works without a CLI flag', async () => {
```

Comment:
This test is copied straight out of https://github.com/nodejs/node/pull/48439

Reply:
Do we have any CLI flags that are boolean _or_ accept a value? If not, then it’s too big a lift to build that functionality since all the CLI stuff is handled in C++.

I’m okay with just accepting that this PR unflags the implementation. We’ll mark it as `dont-backport` when we land it. If we want to release this with the prior `register` PR before waiting for the other Milestone 2 stuff, we can do a quick new PR that either adds a new boolean flag or updates `--experimental-loader` to work without a value.

Reply:
Sorry, I don't mean it should be literally a boolean (not sure if the difference matters). I mean:

```console
node ./main.mjs
# getOptionValue('--experimental-loader') → undefined
```

Un-gate `module.register`:
```console
node --experimental-loader ./main.mjs
# getOptionValue('--experimental-loader') → []
```

Un-gate `module.register` (and register css-loader):
```console
node --experimental-loader=css-loader ./main.mjs
# getOptionValue('--experimental-loader') → ['css-loader']
```

Reply:
What I mean is does our CLI parser allow such a thing? I kind of doubt it.

Regardless, I think we can land this PR and do any flag changes in a follow up.

Reply:
We _do_ have a flag where the value is optional: [`--inspect-brk`](https://nodejs.org/api/cli.html#--inspect-brkhostport). But it’s not an array flag like `--loader`, that can be passed multiple times. I don’t think we have any flags that can be either arrays or booleans.

I think we can push off the flag question to a follow up PR, and mark this as `dont-backport` so that it doesn’t get released and unflag the implementation unless we decide that that’s what we want.

Reply:
SGTM :)